core/src/main/java/org/mskcc/oncotree/crosswalk/MSKConceptCache.java

```diff
@@ -107,21 +126,16 @@ private void resetCache() {
                 topBraidURIsToOncotreeCodes.get(FUSED_CLL_SLL_URI).add("CLL");
             }
         }
+        oncoTreeCodesToMSKConcepts = latestOncoTreeCodesToMSKConcepts;
     }
-
-    private MSKConcept getFromCrosswalkAndSave(String oncoTreeCode) {
-        // only save if we have not seen before (UMLS/NCI info will not be different)
-        if (oncoTreeCodesToMSKConcepts.containsKey(oncoTreeCode)) {
-            return oncoTreeCodesToMSKConcepts.get(oncoTreeCode);
-        }
+    
+    private MSKConcept getFromCrosswalk(String oncoTreeCode) {
```

Comment:
getFromCrosswalk and saving into the cache is now separate -- can be used both for the get() function which saves directly to the existing cache or the resetCache() function.